cmd/cert-checker/main_test.go

```diff
@@ -422,6 +427,31 @@ func TestGetCertsEmptyResults(t *testing.T) {
 	test.AssertNotError(t, err, "Failed to retrieve certificates")
 }
 
+// create emptyDB for testing that null response from mariadb are handled properly
+type emptyDB struct{}
```

Comment:
Optional nit: you can embed a certDB in the emptyDB so that the compiler assumes it has all the same methods, even though they'd cause nil pointer exceptions if they were actually called during the test. This lets you skip defining the `Select` function below.
```suggestion
type emptyDB struct{
	certDB
}
```